src/PKSim.Core/Snapshots/Mappers/OriginDataMapper.cs

```diff
@@ -148,7 +148,7 @@ private double baseParameterValueFrom(Parameter snapshot, IDimension dimension,
          if (snapshot?.Value == null)
             return defaultValue;
 
-         var unit = dimension.Unit(ModelValueFor(snapshot.Unit));
+         var unit = dimension.UnitOrDefault(ModelValueFor(snapshot.Unit));
```

Comment:
In snapshot, if unit is missing, we assume default unit (see tests)

Reply:
>if unit is missing

In which situations it would be the case?

Reply:
Snapshot edited by hand, or code like the one present in the test.
But we can remove it and make it mandatory. Good with that too :)

Reply:
ok, then I would make it mandatory.

Reply:
ok

Reply:
Sounds good. I am going to create a better message than  "Key not found" in that case